app/src/main/java/com/pavesid/androidacademy/ui/MoviesViewModel.kt

```diff
@@ -28,7 +30,27 @@ class MoviesViewModel @ViewModelInject constructor(
     init {
         viewModelScope.launch(Dispatchers.IO + exceptionHandler) {
             _movies.postValue(Resource.loading(null))
-            _movies.postValue(Resource.success(repository.getMovies()))
+            list = repository.getMoviesFromDB().let {
+                if (it.isEmpty()) {
+                    val list = repository.getMovies()
+                    repository.insertMovies(list)
+                    return@let list
+                } else it
+            }
```

Comment:
I think this logic should be inside the repository

Reply:
Done.